cfripper/rules/base_rules.py

```diff
@@ -13,13 +13,51 @@
 specific language governing permissions and limitations under the License.
 """
 import logging
-from typing import List, Set
+from abc import ABC, abstractmethod
+from typing import Dict, List, Optional, Set
 
-from cfripper.model.rule import Rule
+from pycfmodel.model.cf_model import CFModel
+
+from cfripper.config.config import Config
+from cfripper.model.enums import RuleGranularity, RuleMode, RuleRisk
+from cfripper.model.result import Result
 
 logger = logging.getLogger(__file__)
 
 
+class Rule(ABC):
+    RULE_MODE = RuleMode.BLOCKING
+    RISK_VALUE = RuleRisk.MEDIUM
+    GRANULARITY = RuleGranularity.STACK
+
+    def __init__(self, config: Optional[Config]):
+        self._config = config if config else Config()
+
+    @abstractmethod
+    def invoke(self, cfmodel: CFModel, extras: Optional[Dict] = None) -> Result:
+        pass
+
+    def add_failure(
```

Comment:
@jsoucheiron What do you think about moving this logic to Rule constructor? Looks like the best place to have it.